speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/ResearcherMenu.cpp

```diff
@@ -217,6 +289,87 @@ static void SaveSettings(void* /* dummy */)
+/// @param p_param The configuration menu handle
+void InitializeSettings(void* p_param) {
+    // Retrieve all setting variables from the xml file
+    bool* checkboxTasks = new bool[3];
+    checkboxTasks[TASK_LANE_KEEPING] = gfuiMenuGetBoolean(GfParmGetStr(p_param, "dynamic controls/CheckboxTaskLaneKeeping", "checked", NULL), true);
+    checkboxTasks[TASK_SPEED_CONTROL] = gfuiMenuGetBoolean(GfParmGetStr(p_param, "dynamic controls/CheckboxTaskSpeedControl", "checked", NULL), false);
+
+    bool checkboxIndicatorAuditory = gfuiMenuGetBoolean(GfParmGetStr(p_param, "dynamic controls/CheckboxIndicatorAuditory", "checked", NULL), true);
+    bool checkboxIndicatorVisual = gfuiMenuGetBoolean(GfParmGetStr(p_param, "dynamic controls/CheckboxIndicatorVisual", "checked", NULL), true);
+    bool checkboxIndicatorTextual = gfuiMenuGetBoolean(GfParmGetStr(p_param, "dynamic controls/CheckboxIndicatorTextual", "checked", NULL), true);
+
+
+    bool* checkboxInterventions = new bool[5];
+    checkboxInterventions[INTERVENTION_TYPE_NO_SIGNALS] = gfuiMenuGetBoolean(GfParmGetStr(p_param, "dynamic controls/CheckboxTypeNoSignals", "checked", NULL), true);
+    checkboxInterventions[INTERVENTION_TYPE_ONLY_SIGNALS] = gfuiMenuGetBoolean(GfParmGetStr(p_param, "dynamic controls/CheckboxTypeOnlySignals", "checked", NULL), false);
+    checkboxInterventions[INTERVENTION_TYPE_SHARED_CONTROL] = gfuiMenuGetBoolean(GfParmGetStr(p_param, "dynamic controls/CheckboxTypeSharedControl", "checked", NULL), false);
+    checkboxInterventions[INTERVENTION_TYPE_COMPLETE_TAKEOVER] = gfuiMenuGetBoolean(GfParmGetStr(p_param, "dynamic controls/CheckboxTypeCompleteTakeover", "checked", NULL), false);
+
+    bool checkboxInterventionToggle = gfuiMenuGetBoolean(GfParmGetStr(p_param, "dynamic controls/CheckboxPControlInterventionToggle", "checked", NULL), false);
+    bool checkboxPControlGas = gfuiMenuGetBoolean(GfParmGetStr(p_param, "dynamic controls/CheckboxPControlGas", "checked", NULL), true);
+    bool checkboxPControlSteering = gfuiMenuGetBoolean(GfParmGetStr(p_param, "dynamic controls/CheckboxPControlSteering", "checked", NULL), true);
+
+    // Set the max time setting from the xml file
+    m_maxTime = std::stoi(GfParmGetStr(p_param, "dynamic controls/MaxTimeEdit", "default value", NULL));
+    
+
+    // Set the Task settings from the xml file
+    setTask(checkboxTasks);
+
+    // Set the indicator settings from the xml file
+    m_indicators.Audio = checkboxIndicatorAuditory;
+    m_indicators.Icon = checkboxIndicatorVisual;
+    m_indicators.Text = checkboxIndicatorTextual;
+
+    // Set the indicator settings from the xml file
+    m_pControl.ControlInterventionToggle = checkboxInterventionToggle;
+    m_pControl.ControlGas = checkboxPControlGas;
+    m_pControl.ControlSteering = checkboxPControlSteering;
+
+    // Set the participant control settings from the xml file 
+    setInterventionType(checkboxInterventions);
+}
```

Comment:
Every new array[] should have a delete[] array
```suggestion
    setInterventionType(checkboxInterventions);
    
    delete[] checkboxTasks;
    delete[] checkboxInterventions ;
}
```